ls.nim

```diff
@@ -528,59 +542,64 @@ proc checkProject*(ls: LanguageServer, uri: string): Future[void] {.async, gcsaf
     nimsuggest.needsCheckProject = false
     callSoon() do () {.gcsafe.}:
       debug "Running delayed check project...", uri = uri
-      traceAsyncErrors ls.checkProject(uri)
+      # traceAsyncErrors ls.checkProject(uri)
+      asyncCheck ls.checkProject(uri)
```

Comment:
the "better" way of doing this is:
```suggestion
      asyncSpawn ls.checkProject(uri)
```

this should be done in combination with `checkProject` being annotated `{.async: (raises: []).}` to ensure that exceptions are handled _inside_ `checkProject`